.github/workflows/ci.yml

```diff
@@ -14,6 +14,7 @@ jobs:
     steps:
       - uses: actions/checkout@v2
       - uses: dtolnay/rust-toolchain@1.36.0
+      - run: cargo fmt --all -- --check
```

Comment:
would it not be better to add a separate CI target for rustfmt? and why use the MSRV for running the formatting?

something like-

```yaml
  fmt:
    name: format
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3
      - uses: dtolnay/rust-toolchain@nightly
        with:
          components: rustfmt
      - uses: actions-rs/cargo@v1
        with:
          command: fmt
          args: --all -- --check
```

Reply:
not sure if introducing yet another toolchain is worth here. You can use the stable chain instead, which already exists.

But I'm down for using checkout@3 as well as dtolnay's rust-toolchain.
I have a weird connection with actions-rs, because some of their actions are no longer maintained, but cargo looks good to me, so yeah, that's worth a rewrite in the github workflow.

Reply:
> not sure if introducing yet another toolchain is worth here

in what sense? I would personally split these into separate targets even if they both used the stable toolchain